### ISSUE TITLE
Enable CS8073 in a warning wave

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#45744]
CS8073 applies in a scenario such as this:

```csharp
class Program
{
    public static void M(S s)
    {
        if (s == null) { } // CS8073
        if (s != null) { } // CS8073
    }
}
struct S
{
    public static bool operator==(S s1, S s2) => false;
    public static bool operator!=(S s1, S s2) => true;
    public override bool Equals(object other) => false;
    public override int GetHashCode() => 0;
}
```

```none
(6,13): warning CS8073: The result of the expression is always 'false' since a value of type 'S' is never equal to 'null' of type 'S?'
(7,13): warning CS8073: The result of the expression is always 'true' since a value of type 'S' is never equal to 'null' of type 'S?'
```

The compiler currently only produces this warning in "strict" mode. As part of the [warning wave feature](https://github.com/dotnet/roslyn/issues/45701), this warning should be included in warning wave "5".
